src/renderer/qt/qt_window_manager.cc

```diff
@@ -341,16 +341,61 @@ void FillCandidates(const commands::Candidates &candidates,
 }
 }  // namespace
 
+struct VirtualRect {
+  const QScreen *screen;
```

Comment:
Can’t we simply store two `Rect` values instead of `QScreen *`, whose lifetime is not obvious to future readers?
```
  Rect screen_rect;
  Rect rect;  // perhaps we should use a bit more descriptive name?
```

You should be able to initialize `screen_rect` with `GetRect(screen->geometry())`, and once it’s initialized we do not need `screen` any more.

Reply:
VirtualRect now holds `screen_rect`  instead of a QScreen object.

I keep `rect` as is. Maybe we might need to rename `VirtualRect` itself.

Reply:
One idea is changing VirtualRect to VirtualPreeditPositions